### ISSUE TITLE
Issue on Bag Change: Changed New Appraisal Type

### ISSUE THREAD

**Issue by danloveg** [NationalCentreTruthReconciliation/Secure-Record-Transfer#18]
When an admin opens a `Bag` from the administrator site and clicks either of:

- Generate Zipped Bag
- Where is This Bag?
- View Metadata as HTML

A history item gets logged that says "Changed New Appraisal Type," even if the admin does not change the appraisal type. What I would really like is to be able to add an appraisal statement using a separate form from the `Bag` change form, the same way you can update a user's password using a separate change form.

If this is too difficult, it should be possible to fix the History issue by setting the appraisal type to empty initially. There is no empty choice in the `Bag` change form for appraisal type, so adding the empty item would be simple.

Related to this issue, the three links (Generate Zipped Bag, etc... ) should not trigger a `Bag` change event. These links do not update the model in the database and are therefore not _changing_ the `Bag`.

**Comment by whikloj** [comment]
I can work this one. But I have some questions to make sure I understand the use cases.

It appears like you can only add new archival/monetary appraisals but you can't remove them. Is that by design?

If not, would it be okay to move them to a separate model and provide a spot to edit/remove attached appraisals? 

We could still update the bag caais JSON to include the statements, but by having them external then if someone wants to edit or remove one. We'd just remove that whole section from the JSON and re-serialize the remaining appraisals.

**Comment by danloveg** [comment]
I built the appraisal part haphazardly, I'm not surprised I forgot something like being able to remove one. Yes, I think the appraisal should be a separate model, and it'd be ideal if someone could click a button on the `Bag` page to add a new appraisal.

If a new Appraisal model is made, the Bag edit form should have an inline form for related appraisals, similar to how a `BagGroup` has an inline form. We might want to create a taxonomy later for the types of appraisal as well, in case someone wants to make other types of appraisals than monetary/archival. This is low priority, though.

I agree that saving an Appraisal should be tied to the appraisal section in the Bag's JSON. When making a change to an appraisal, the changes will also need to be propagated to the Bag itself on the filesystem.

**Comment by danloveg** [comment]
Addressed by #54

A new `Appraisal` model was made, and a new `Submission` model was made to connect appraisals and bags. Appraisals can be added inline from an existing `Submission`. When an appraisal is added to a submission, the BagIt bag on the file system associated with the submission is updated. Also, when an appraisal is removed, the BagIt bag on the file system is updated accordingly.

The level of detail, review status, and accession identifier have been extracted to the Submission model, rather than the Bag model. The Bag model basically only needs only to track the metadata and the location of the bag, so these fields did not fit well in the bag model.

The history issues I mentioned when this issue was opened (i.e., "Changed new Appraisal Type") have also been addressed. I have created a new *view* for the `Submission` that allows you to view the transfer report. The form is no longer submitted when the transfer report is viewed, so no extra history item is logged. Similarly, the generate downloadable bag link has been converted to a *view* as well, so no polluted history there either.

The location of the bag has been added directly on the Bag change page, so no extra clicking needs take place for an admin to find the location of the bag.
